src/PHPMailer.php

```diff
@@ -1942,33 +1942,33 @@ public function smtpConnect($options = null)
         foreach ($hosts as $hostentry) {
             $hostinfo = [];
             if (!preg_match(
-                '/^((ssl|tls):\/\/)*([a-zA-Z\d.-]*|\[[a-fA-F\d:]+]):?(\d*)$/',
+                '/^(?:(ssl|tls):\/\/)?(.+?)(?:(\d*))?$/',
```

Comment:
The initial non-capturing group change is a good idea, but the rest of it loses most of the basic filtering for things that look vaguely domain-like, for example by allowing `exam ple.com`. I think there may well be a problem with the original matching on the port number, in that it will match trailing numbers even if they're not preceded by a `:`, but I don't think this new version deals with that either - for example it messes up on `example.com123`. To be fair, I'm not sure that domains ending in numbers are valid, but port numbers *definitely* need a preceding `:`. This needs lots of test cases!

Reply:
Sharp, regex should have been `'/^(?:(ssl|tls):\/\/)?(.+?)(?::(\d*))?$/'`, fixed that now

Reply:
That's good, thanks - I still don't see value in downgrading the match from `([a-zA-Z\d.-]*|\[[a-fA-F\d:]+])` to the much broader `(.+?)` - that will match pretty much anything at all, so it's not much use as a validation. It could be shortened by using the `/i` pattern modifier I guess.

Reply:
The reason I did, is because the "is a host valid"-responsibility now **only** lies in the isValidHost method:

a) Method isValidHost is public/static; checking the validness of a host for an outside caller isn't the complete set of checks that are performed within PHPMailer.
b) One could now derive from this class and only overload the isValidHost method to implement it's own check(s).

Do you agree that moving the `([a-zA-Z\d.-]*|\[[a-fA-F\d:]+])` regex to the isValidHost method will be a good solution?

Reply:
Did just that.

Reply:
I agree that moving responsibility there is a good idea. While we're there, checks like:

            return (bool) filter_var(trim($host, '[]'), FILTER_VALIDATE_IP, FILTER_FLAG_IPV6);

are probably better done as:

            return filter_var(trim($host, '[]'), FILTER_VALIDATE_IP, FILTER_FLAG_IPV6) !== false;

Though I spot a potential hole there too - `[[<ipv6>]]` would pass this test.

Reply:
Fixed. See latest commits

Reply:
Hmmmm, this has a side effect. See https://travis-ci.org/PHPMailer/PHPMailer/jobs/621083134?utm_medium=notification&utm_source=github_status